CONTRIBUTING.md

```diff
@@ -64,8 +64,29 @@ Instructions for adding style to a component:
 We have adopted the [ducks pattern](https://www.freecodecamp.org/news/scaling-your-redux-app-with-ducks-6115955638be/) for structuring our application.
 
 ### Testing
+We are using [Jest](https://jestjs.io/) as our testing framework foundation along with [Testing Library](https://testing-library.com/) for testing UI components.
+
+#### Tests and UI Components
+We adhere to the principles of the [Testing Library](https://testing-library.com/docs/guiding-principles) framework
+because it give us more confidence in validating the application as close as possible to the user.
+
+Which also means we're more focusing in testing complex UI components instead of *leaf* componnets.
+The latter are more likely to change, making tests more brittle and creating unncessary overhead.
+
+Therefore there is no need to test each *leaf* component, unless it brings any specific value.
+
+UI *Component Unit Tests* cover *non connected* components.
+While *Integration Tests* cover "connected" components.
+
+*Note*: Testing UI components this way is not considered *pure unit testing* and we're fine with that, again it's all about the confidence tests provide instead of a *false sense* of code coverage.
+
+To stay close to simulating end-user interaction with the application it's important to query DOM elements being visible on a page instead not pull internal components or nodes. Please get familiar with the queries priority order from [Using queries](https://testing-library.com/docs/guide-which-query).
```

Comment:
page instead not pull internal components or nodes.

->

 page instead of pulling internal components or nodes